packages/astro/components/ViewTransitions.astro

```diff
@@ -0,0 +1,137 @@
+---
+type Fallback = 'none' | 'animate' | 'swap';
+
+export interface Props {
+	fallback?: Fallback;
+}
+
+const { fallback = 'animate' } = Astro.props as Props;
+---
+<meta name="astro-view-transitions-enabled" content="true">
+<meta name="astro-view-transitions-fallback" content={fallback}>
+<script>
+	type Fallback = 'none' | 'animate' | 'swap';
+	type Direction = 'forward' | 'back';
+
+	let currentIndex = history.state?.index || 0;
+	if(!history.state) {
+		history.replaceState({index: currentIndex}, document.title);
+	}
+
+	const supportsViewTransitions = !!document.startViewTransition;
+	const transitionEnabledOnThisPage = () => !!document.querySelector('[name="astro-view-transitions-enabled"]');
+
+  async function getHTML(href: string) {
+    let res = await fetch(href)
+    let html = await res.text();
+		return { ok: res.ok, html };
+  }
+
+	function getFallback(): Fallback {
+		let el = document.querySelector('[name="astro-view-transitions-fallback"]');
+		if(el) {
+			return el.getAttribute('content') as Fallback;
+		}
+		return 'animate';
+	}
+
+	const parser = new DOMParser();
+
+  async function updateDOM(dir: Direction, html: string, fallback: Fallback = 'swap') {
+    const doc = parser.parseFromString(html, 'text/html');
+		doc.documentElement.dataset.astroTransition = dir;
+		const swap = () => document.documentElement.replaceWith(doc.documentElement);
+
+		if(fallback === 'animate') {
+			let isAnimating = false;
+			addEventListener('animationstart', () => isAnimating = true, { once: true });
+
+			document.documentElement.dataset.astroTransitionFallback = 'old';
+			doc.documentElement.dataset.astroTransitionFallback = 'new';
+			// Trigger the animations
+			addEventListener('animationend', swap, { once: true });
+			// If there are no animations, go ahead and swap on next tick
+			setTimeout(() => !isAnimating && swap());
+		} else {
+			swap();
+		}
+  }
+
+	async function navigate(dir: Direction, href: string) {
+		let htmlPromise = getHTML(href);
+		let finished: Promise<void>;
+		const { html, ok } = await htmlPromise;
```

Comment:
```suggestion
		const finished: Promise<void>;
		const { html, ok } = await getHTML(href);
```

We don't use `htmlPromise` anywhere else.

Reply:
Finished is used though, it needs to be a `let`.